src/app/modules/main/chat_section/module.nim

```diff
@@ -929,6 +933,10 @@ proc updateTokenPermissionModel*(self: Module, permissions: Table[string, CheckP
       var permissionSatisfied = true
 
       for index, tokenCriteriaItem in tokenPermissionItem.getTokenCriteria().getItems():
+        let criteriaMet = criteriaResult.criteria[index]
+
+        if tokenCriteriaItem.criteriaMet == criteriaMet:
```

Comment:
What if the criteria is still met but other params of the permission have changed?

Reply:
`permissions: Table[string, CheckPermissionsResultDto]` only contains the criterias, so only the boolean saying if the criteria is updated.

Long story short, this function is only called with the permission results, not when a permission is updated. That's handled elsewhere.